src/utils/legacyLayerParamUtils.js

```diff
@@ -173,3 +173,94 @@ export async function getKmlLayerFromLegacyAdminIdParam(adminId) {
         kmlMetadata,
     })
 }
+
+/**
+ * Parse all params to check if a layer-id=features-ids parameter has been sent. If this is the
+ * case, we either add the layers parameter if it doesn't exist, add the layer to the existing
+ * layers param if it exists and the layer is not part of it, or we need to insert the features to
+ * the existing layer
+ *
+ * @param {Object} params The parameters sent to the legacy router, as a key - value pair object
+ * @param {Store} store
+ * @param {Query} newQuery
+ */
+export function handleLegacyFeaturePreSelectionParam(params, store, newQuery) {
+    params.forEach((param_value, param_key) => {
+        if (store.state.layers.config.some((layer) => layer.id === param_key)) {
+            const featuresIds = param_value.split(',').join(':')
+            if (newQuery.layers?.includes(param_key)) {
+                // the layer given as key is also in the query 'layers'
+                // we need to ensure all params are kept intact
+                newQuery.layers = createLayersParamForFeaturePreselection(
+                    param_key,
+                    param_value,
+                    newQuery.layers
+                )
+            } else if (newQuery.layers) {
+                // there is a 'layers' query, but the layer given as key is not part of it,
+                // we simply add the layer and its features at the end of the existing query
+                newQuery.layers = newQuery.layers + `;${param_key}@features=${featuresIds}`
+            } else {
+                // there was not 'layers' param within the query, we create one for this layer.
+                newQuery.layers = `${param_key}@features=${featuresIds}`
+            }
+        }
+    })
+}
+
+/**
+ * @param {String} layerId The layer Id for which we have features
+ * @param {String} featuresIds The features ids we need to add as a parameter. This a coma-separated
+ *   string
+ * @param {String} layers The new Query layers parameter, a semicolon separated string
+ * @returns
+ */
+export function createLayersParamForFeaturePreselection(layerId, featuresIds, layers) {
+    // if the features Ids are null, we replace them with an empty array
+    // we also remove all empty strings from the featuresIds
+    const featuresArray = (featuresIds ? featuresIds.split(',') : []).filter(
+        (featureId) => featureId !== ''
+    ) // we go through each layer
+    console.log(featuresArray)
+    const layersArray = layers.split(';')
+    for (let layerIndex = 0; layerIndex < layersArray.length; layerIndex++) {
```

Comment:
This should be simplified. This code has way too much indentation which make it hard to read/understand, it needs to be break down.
I don't think we need to loop over all layers we need to find the one that match `layerId` something like this

```javascript
const layer = layers.split(';').find(l => l.match(new RegExp(`^${layerId}(@.*)?$`)))
```